data/battle_scripts_1.s

```diff
@@ -440,6 +440,17 @@ gBattleScriptsForMoveEffects::
 	.4byte BattleScript_EffectSyrupBomb               @ EFFECT_SYRUP_BOMB
 	.4byte BattleScript_EffectHit                     @ EFFECT_IVY_CUDGEL
 	.4byte BattleScript_EffectMaxMove                 @ EFFECT_MAX_MOVE
+	.4byte BattleScript_EffectGlaiveRush              @ EFFECT_GLAIVE_RUSH
+
+BattleScript_EffectGlaiveRush::
+	call BattleScript_EffectHit_Ret
+	jumpifhalfword CMP_COMMON_BITS, gMoveResultFlags, MOVE_RESULT_DOESNT_AFFECT_FOE, BattleScript_GlaiveRushNoEffect
```

Comment:
No need to make a new battle script, just jump to the relevant part of EffectHit

Reply:
Yup, looks better. You could even just jump to the battle script of `tryfaintmon BS_TARGET` in `EffectHit` and save a line of code.

Reply:
Yeah, uou can just add a new label in the EffectHit battle script at that point and jump to there

Reply:
Question, would there be any harm in moving `seteffectwithchance` to `BattleScript_EffectHit_Ret`? Not every BS that uses `BattleScript_EffectHit_Ret` has `seteffectwithchance` in them but most do. That would clean up a lot of lines.

Reply:
Weren't that many lines in the end but it still looks better.

Reply:
Much better, looks like you're done if your tests all pass :)

Reply:
Yeah, all pass :sweat_smile: 

Thank you for you help!